hackathon_site/dashboard/frontend/src/pages/Orders/Orders.test.tsx

```diff
@@ -1,40 +1,52 @@
+        store = makeStoreWithEntities({
+            hardware: mockHardware,
+            allOrders: mockPendingOrders,
+        });
+    });
     test("Has necessary page elements", () => {
-        const { getByText, getByTestId } = render(<Orders />);
+        const { getByText, getByTestId } = render(<Orders />, { store });
 
         expect(getByText("Orders")).toBeInTheDocument();
         expect(getByTestId("ordersCountDivider")).toBeInTheDocument();
     });
-    // TODO: after newhacks, implement this test using the new thunk and not mock data
-    // test("Display correct order cards", () => {
-    //     const { getByText, getByTestId } = render(<Orders />);
-    //     mockPendingOrders.forEach((order) => {
-    //         const orderItem = getByTestId(`order-item-${order.id}`);
-    //         const orderStatus = order.status;
-    //         const orderDetails = within(orderItem);
-    //
-    //         if (orderStatus === "Submitted") {
-    //             const date = new Date(order.created_at);
-    //             const month = date.toLocaleString("default", { month: "short" });
-    //             const day = date.getDate();
-    //             const hoursAndMinutes = date.getHours() + ":" + date.getMinutes();
-    //
-    //             expect(orderDetails.getByText(order.id)).toBeInTheDocument();
-    //             expect(orderDetails.getByText(order.items.length)).toBeInTheDocument();
-    //             expect(orderDetails.getByText(order.team_code)).toBeInTheDocument();
-    //             expect(
-    //                 orderDetails.getByText(`${month} ${day}, ${hoursAndMinutes}`)
-    //             ).toBeInTheDocument();
-    //         } else {
-    //             expect(
-    //                 orderDetails.getByText("Order Card Component")
-    //             ).toBeInTheDocument();
-    //         }
-    //     });
-    // });
+
+    test("Display correct order cards", () => {
+        const { getByTestId } = render(<Orders />, { store });
+        mockPendingOrders.forEach((order) => {
+            const orderStatus = order.status;
+
+            if (orderStatus === "Submitted" || orderStatus === "Ready for Pickup") {
+                const orderItem = getByTestId(`order-item-${order.id}`);
+                const orderDetails = within(orderItem);
+                const date = new Date(order.created_at);
+                const month = date.toLocaleString("default", { month: "short" });
+                const day = date.getDate();
+                const hoursAndMinutes = date.getHours() + ":" + date.getMinutes();
+
+                expect(orderDetails.getByText(order.id)).toBeInTheDocument();
+                expect(orderDetails.getByText(order.items.length)).toBeInTheDocument();
+                expect(orderDetails.getByText(order.team_code)).toBeInTheDocument();
+                expect(
+                    orderDetails.getByText(`${month} ${day}, ${hoursAndMinutes}`)
+                ).toBeInTheDocument();
```

Comment:
`expect(orderDetails.getByText(order.status)).toBeInTheDocument();` And remove the if statement above

Reply:
These tests are specifically for pending orders. Tests for other statuses can be completed in the future(refer to TODO below)
@Leo6Leo